mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewInternalViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.airbnb.mvrx
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import java.util.UUID
+import java.util.concurrent.ConcurrentHashMap
+
+class MavericksViewInternalViewModel(state: SavedStateHandle): ViewModel() {
+    val lastDeliveredStates: ConcurrentHashMap<String, Any> = state[PERSISTED_LAST_DELIVERED_STATES_KEY] ?: ConcurrentHashMap<String, Any>().also {
```

Comment:
mark these properties all as internal? It's a bit unfortunate that the class is public, but I see why you had to do that

Reply:
is this a new feature you are adding to save the states and subscriptions across process death?

it seems like a good thing to have, but also seems like something we need to be careful about to get right. Are you sure that the ConcurrentHashMap can be saved and retrieved out of the saved state correctly? it should be serializable, but a quick look at the SavedStateHandle source seems like it needs to be a Parcelable

Reply:
hmm the docs https://developer.android.com/topic/libraries/architecture/viewmodel-savedstate#non-parcelable seem to imply both would work. the perusing the code, validate does check for acceptable type of which serializable is one